src/System.IO/tests/StreamReader/StreamReaderTests.cs

```diff
@@ -0,0 +1,379 @@
+        {
+            var baseInfo = GetCharArrayStream();
+            var sr = baseInfo.Item2;
+
+            var chArr = new Char[baseInfo.Item1.Length];
+
+            var read = sr.Read(chArr, 0, chArr.Length);
+
+            Assert.Equal(chArr.Length, read);
+            for (int i = 0; i < baseInfo.Item1.Length; i++)
+            {
+                Assert.Equal(baseInfo.Item1[i], chArr[i]);
+            }
+        }
+
+        [Fact]
+        public static async Task VanillaReads2WithAsync()
+        {
+            var baseInfo = GetCharArrayStream();
+
+            var sr = baseInfo.Item2;
+
+            var chArr = new Char[baseInfo.Item1.Length];
+
+            var read = await sr.ReadAsync(chArr, 4, 3);
+
+            Assert.Equal(read, 3);
+            for (int i = 0; i < 3; i++)
+            {
+                Assert.Equal(baseInfo.Item1[i], chArr[i + 4]);
+            }
+        }
+        [Fact]
+        public static void ObjectDisposedReadLine()
+        {
+            var baseInfo = GetCharArrayStream();
+            var sr = baseInfo.Item2;
+
+            sr.Dispose();
+            Assert.Throws<ObjectDisposedException>(() =>
+            {
+                sr.ReadLine();
+            });
+        }
+        [Fact]
+        public static void ObjectDisposedReadLineBaseStream()
+        {
+            var ms = GetLargeStream();
+            var sr = new StreamReader(ms);
+
+            ms.Dispose();
+            Assert.Throws<ObjectDisposedException>(() =>
+            {
+                sr.ReadLine();
+            });
+        }
+       
+        [Fact]
+        public static void VanillaReadLines()
+        {
+            var baseInfo = GetCharArrayStream();
+            var sr = baseInfo.Item2;
+
+            String valueString = new String(baseInfo.Item1);
+
+
+            var data = sr.ReadLine();
+            Assert.Equal(valueString.Substring(0, valueString.IndexOf('\r')), data);
+
+            data = sr.ReadLine();
+            Assert.Equal(valueString.Substring(valueString.IndexOf('\r') + 1, 3), data);
+
+            data = sr.ReadLine();
+            Assert.Equal(valueString.Substring(valueString.IndexOf('\n') + 1, 2), data);
+
+            data = sr.ReadLine();
+            Assert.Equal((valueString.Substring(valueString.LastIndexOf('\n') + 1)), data);
+        }
+
+        [Fact]
+        public static void VanillaReadLines2()
+        {
+            var baseInfo = GetCharArrayStream();
+            var sr = baseInfo.Item2;
+
+            String valueString = new String(baseInfo.Item1);
+
+            var temp = new char[10];
+            sr.Read(temp, 0, 1);
+            var data = sr.ReadLine();
+            Assert.Equal(valueString.Substring(1, valueString.IndexOf('\r') - 1), data);
+        }
+
+        [Fact]
+        public static async Task ContinuousNewLinesAndTabsAsync()
+        {
+            var ms = new MemoryStream();
+            var sw = new StreamWriter(ms);
+            sw.Write("\n\n\r\r\n");
```

Comment:
It feels ok about it.  StreamReader treats all of '\r', '\n', and "\r\n" as newline endings in ReadLine{Async}, regardless of platform.